keps/sig-storage/625-csi-migration/README.md

```diff
@@ -156,10 +163,209 @@ Running this set of tests is optional for a per-provider basis. We would
+To turn it on by default in Beta, we require each in-tree plugin to at least manually test the upgrade->downgrade->upgrade path.
+For GA, we require such test exists in each driver's test CI.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+There will not be API removal in CSI migration itself. But eventually when CSI migration is all finished. We will plan to remove all in-tree plugins.
+So we will have in-tree plugin deprecated when CSIMigration{cloud-provider} goes to beta. And code removal will be required eventually.
+In addition, some CSI drivers are not able to maintain 100% backwards compatibility, so those drivers need to deprecate certain behaviors. For example, azurefile and vSphere has some corner cases that will not work anymore. (TODO: add more details.)
+
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+  We will have metrics `csi_sidecar_duration_seconds` on the CSI sidecars and `storage_operation_duration_seconds` on the kube-controller-manager and kubelet side to indicate whether this operation is a migrated operation or not. These metrics will have a `migrated` field to indicate if this is a migrated operation.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+  - [x] Metrics
+    - Metric name: csi_sidecar_duration_seconds && storage_operation_duration_seconds, these metrics will have a `migrated` field
+    - [Optional] Aggregation method:
+    - Components exposing the metric: CSI sidecars, kubelet, kube-controller-manager
+  - [x] Other (treat as last resort)
+    - Details: Pod using in-tree plugin has failure.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  - SLO with migration on matches the existing plugin's in-tree SLO with offset less than 1%
+
+* **Are there any missing metrics that would be useful to have to improve observability
+of this feature?**
+Node side CSI operation metrics. It will be implemented in the GA phase.
+
+### Dependencies
+
+
+* **Does this feature depend on any specific services running in the cluster?**
+
+  - Corresponding CSI Driver for in-tree CSI migration enabled plugin
+    - Usage description:
+      - Impact of its outage on the feature: in-tree plugin stops working without the CSI Driver properly setup
+      - Impact of its degraded performance or high-error rates on the feature: Error or performance decrease will be reflected for volume intensive operation
+
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+  Yes.
+  - For volume attach/detach:
+    VolumeAttachment CREATE/PATCH APIs will be called for volume attachment by kube-controller-manager and csi-attacher. VolumeAttachment delete API will be called for volume detachment by csi-attacher. 1 for each needed.
```

Comment:
csi-attacher does not make CREATE calls.

Reply:
Thanks for clarifying, I changed the section.